manifests/server/puppetserver.pp

```diff
@@ -0,0 +1,180 @@
+# [*master_var_dir*]        - (optional) path to puppet var dir
+# [*enable_profiler*]       - (optional) enable or disable profiling for the Ruby code
+#      (true|false)
+# [*allow_header_cert_info  - (optional) Allows the "ssl_client_header" and
+#      (true|false)           "ssl_client_verify_header" options set in
+#                             puppet.conf to work. These headers will be
+#                             ignored unless "allow-header-cert-info" is true
+
+# puppetserver.conf is in HOCON format, which is a superset of JSON:
+# - https://github.com/puppetlabs/puppet-server/blob/master/documentation/configuration.markdown
+# - https://github.com/typesafehub/config#using-hocon-the-json-superset
+# puppet-puppet will use https://github.com/puppetlabs/puppetlabs-hocon to
+# manage hocon settings because that appears to be the approach the puppetserver
+# developers have chosen to support
+
+class puppet::server::puppetserver (
+  # aside from memory (defaults to 2g), these are the puppetserver defaults
+  # Setting them here makes it easier to anticipate behavior
+  $enabled                = true,
+  $memory_pct             = 70,
+  $memory                 = undef,
+  $max_active_instances   = $::processorcount + 2,
+  $logging_config         = $puppet::params::puppetserver_logback_conf,
+  $gem_home               = $puppet::params::puppetserver_gem_home,
+  $master_conf_dir         = $puppet::params::puppet_confdir,
+  $master_var_dir         = $puppet::params::puppet_vardir,
+  $enable_profiler        = false,
+  $allow_header_cert_info = false,
+  $bootstrap_cfg          = $puppet::params::puppetserver_bootstrap_conf,
+) {
+
+  include puppet
+  include puppet::server
+
+  # Calculate JVM memory based on percentage if specified
+  # Should JVM settings be their own class?
+  if ($memory_pct != undef) and ($memory != undef) {
+    fail('memory and memory_pct cannot both be set at the same time')
+  }
+  if ($memory_pct != undef) and ($memory == undef) {
+    $rounded_mem = floor($::memorysize_mb * $memory_pct * 0.01)
+    $jvm_memory = "${rounded_mem}m"
+  }
+  if ($memory_pct == undef) and ($memory != undef) {
+    $jvm_memory = $memory
+  }
+
+  $service_ensure = $enabled? {
+    true    => 'running',
+    default => 'stopped',
+  }
+
+  Ini_subsetting {
+    ensure            => present,
+    section           => '',
+    key_val_separator => '=',
+    quote_char        => '"',
+    path              => $puppet::params::server_service_conf,
+    setting           => 'JAVA_ARGS',
+    notify            => Service[$puppet::params::server_service],
+  }
+
+  ini_subsetting {'puppetserver_xmx_memory':
+    subsetting        => '-Xmx',
+    value             => $jvm_memory,
+  }
+  ini_subsetting {'puppetserver_xms_memory':
+    subsetting        => '-Xms',
+    value             => $jvm_memory,
+  }
+  ini_subsetting {'puppetserver_maxpermsize':
+    subsetting        => '-XX:MaxPermSize=',
```

Comment:
I'm pretty ignorant of java version differences; what would we need to do for java 8? I'm simply re-implementing the stock puppetserver configuration here. I'd be glad to switch based on java version, or to parameterize it.